webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -603,6 +385,347 @@ private static void addScrambles(PdfWriter docWriter, Document doc, ScrambleRequ
+        rules.add("Your solution must be at most " + maxMoves + " moves, including rotations.");
+        rules.add("Your solution must not be directly derived from any part of the scrambling algorithm.");
+        ct.addElement(rules);
+        MAGIC_NUMBER = 150; // kill me now
+        ct.setSimpleColumn(left+padding, scrambleBorderTop, rulesRight-padding, rulesTop-MAGIC_NUMBER, 0, Element.ALIGN_LEFT);
+        ct.go();
+
+        doc.newPage();
+    }
+
+    private static void addGenericFmcSolutionSheet(PdfWriter docWriter, Document doc, String globalTitle) throws DocumentException, IOException {
+        addFmcSolutionSheet(docWriter, doc, null, globalTitle, -1);
+    }
+
+    private static void addFmcScrambleCutoutSheet(PdfWriter docWriter, Document doc, ScrambleRequest scrambleRequest, String globalTitle, int index) throws DocumentException, IOException {
+        Rectangle pageSize = doc.getPageSize();
+        String scramble = scrambleRequest.scrambles[index];
+        PdfContentByte cb = docWriter.getDirectContent();
+
+        BaseFont bf = BaseFont.createFont(BaseFont.HELVETICA, BaseFont.CP1252, BaseFont.NOT_EMBEDDED);
+        BaseFont bfBold = BaseFont.createFont(BaseFont.HELVETICA_BOLD, BaseFont.CP1252, BaseFont.NOT_EMBEDDED);
+
+        int bottom = 30;
+        int left = 35;
+        int right = (int) (pageSize.getWidth()-left);
+        int top = (int) (pageSize.getHeight()-bottom);
+
+        int height = top - bottom;
+        int width = right - left;
+
+        int fontSize;
+        int padding = 90;
+        int marginBottom = 10;
+        int offsetTop = 27;
+
+        int availableScrambleSpace = width - padding;
+        int scrambleFontSize = 20;
+        float scrambleWidth;
+        do {
+            scrambleFontSize--;
+            scrambleWidth = bf.getWidthPoint(scramble, scrambleFontSize);
+        } while (scrambleWidth > availableScrambleSpace);
+
+        int availableScrambleWidth = (int) (width * .45);
+        int availableScrambleHeight = height - (int) (height * .77) - 90;
+        Dimension dim = scrambleRequest.scrambler.getPreferredSize(availableScrambleWidth - 8, availableScrambleHeight - 8);
+        PdfTemplate tp = cb.createTemplate(dim.width, dim.height);
+        Graphics2D g2 = new PdfGraphics2D(tp, dim.width, dim.height, new DefaultFontMapper());
+
+        try {
+            Svg svg = scrambleRequest.scrambler.drawScramble(scramble, scrambleRequest.colorScheme);
+            drawSvgToGraphics2D(svg, g2, dim);
+        } catch (InvalidScrambleException e) {
+            l.log(Level.INFO, "", e);
+        } finally {
+            g2.dispose();
+        }
+
+        cb.setLineDash(3f, 3f);
+        cb.moveTo(left, top - offsetTop + (marginBottom * 3));
+        cb.lineTo(right, top - offsetTop + (marginBottom * 3));
+        cb.stroke();
+
+        final int scramblesPerSheet = 8;
+        for (int y = 0; y < scramblesPerSheet; y++) {
+            cb.beginText();
+            fontSize = 12;
```

Comment:
It's not safe to assume that 12 will always fit. We should probably add some newlines and dynamically compute the font size. However, this PR has been dragging on for too long, so let's get this merged up, and we can tackle this in a later PR if necessary. I've created https://github.com/thewca/tnoodle/issues/235 to track this. Can you just add a comment here saying something like

> We've hardcoded this to 12 for now, see https://github.com/thewca/tnoodle/issues/235 for discussion about a better solution.